pkg/controller/registry/registry_controller_test.go

```diff
@@ -18,70 +18,298 @@
+	if err != nil {
+		return job, err
+	}
+
+	registryAddress := kubicutil.SafeID(registry.Spec.HostPort)
+	secret, err := registry.GetCertificateSecret(c)
+	if err != nil {
+		return job, err
+	}
+
+	nodes, err := getAllNodes(c)
+	if err != nil {
+		return job, err
+	}
+	numNodes:= len(nodes)
+
+	dockerDstDir := filepath.Join(dockerCertsDir, registry.Spec.HostPort)
+	podmanDstDir := filepath.Join(podmanCertsDir, registry.Spec.HostPort)
+
+	cmdTemplate := "if [[ ! -f '%s/ca.crt' ]]; then echo 'cert not found at %s'; exit 1; fi "
+	commands := []string {
+		fmt.Sprintf(cmdTemplate, dockerDstDir, dockerDstDir),
+		fmt.Sprintf(cmdTemplate,podmanDstDir, podmanDstDir),
+	}
+
+	job, err = getRunnerJobWithSecrets(&runnerWithSecrets{
+		Commands:     []string{strings.Join(commands, " ; ")},
+		JobName:      checkJobKey.Name,
+		NumNodes:     int32(numNodes),
+		JobNamespace: metav1.NamespaceSystem,
+		Secrets: map[string]*corev1.Secret{
+		},
+		Labels: map[string]string{
+			jobInstallLabelHostPort: registryAddress,
+			jobInstallLabelHash:     getSecretHash(secret),
+		},
+		HostPaths: []string{
+			"/etc/docker",
+			"/etc/containers",
+		},
+		AntiAffinity: map[string]string{
+			jobInstallLabelHostPort: registryAddress,
+		},
+	})
+
+		return job, err
+}
+
+func waitRegistryCreated(c client.Client, registryName string, timeout time.Duration) error{
+
+	registry := &kubicv1beta1.Registry{}
+	var err error
+	wait.Poll(100*time.Millisecond, timeout, func() (bool, error) {
+		err = c.Get(context.TODO(), types.NamespacedName{Name: registryName}, registry)
+
+		if err != nil && !apierrors.IsNotFound(err) {
+			return false, err
+		}
+
+		return true, nil
+	})
+	return err
+}
+
+func waitRegistryUpdates(c client.Client, registryName string, expected int32, timeout time.Duration) (int32, error){
+
+	registry := &kubicv1beta1.Registry{}
+	numNodes := int32(-1)
+	var err error
+	wait.Poll(100*time.Millisecond, timeout, func() (bool, error) {
+		err = c.Get(context.TODO(), types.NamespacedName{Name: registryName}, registry)
+		if err == nil {
+			numNodes = int32(registry.Status.Certificate.NumNodes)
+		} else if !apierrors.IsNotFound(err) {
+			return false, err
+		}
+		err = nil
+		return (numNodes == expected), nil
+	})
+	return numNodes, err
+}
+
+func waitJobCompletations(c client.Client, jobName string, expected int32, timeout time.Duration) (int32, error){
+
+	job := &batchv1.Job{}
+	numNodes := int32(0)
+	var err error
+	wait.Poll(100*time.Millisecond, timeout, func() (bool, error) {
+		err = c.Get(context.TODO(), types.NamespacedName{Name: jobName, Namespace: metav1.NamespaceSystem}, job)
+		if err == nil {
+			numNodes = job.Status.Succeeded
+		} else	if !apierrors.IsNotFound(err) {
+			return false, err
+		}
+		err = nil
+		return numNodes == expected, nil
+
+	})
+	return numNodes, err
+}
 
-const timeout = time.Second * 5
 
 func TestReconcile(t *testing.T) {
 
-	test.SkipUnlessIntegrationTesting(t)
+	var err error
```

Comment:
we can remove this and by line 276 we can use `err = addRegController(mgr,newRegistryReconcilier(mgr))`

Reply:
As the variable is used in many places, when adding new error check I found I have to change when it was first declared. I think that this makes clear it is a global variable for the whole function.

Reply:
In golang is a tacit consensus to reuse same var err everywhere without to declare it like that. It is a common idiom, checkout other projects etc